lib/bibliothecary/parsers/pypi.rb

```diff
@@ -102,7 +102,15 @@ def self.parse_pyproject(file_contents, options: {})
         # Parse poetry [tool.poetry] deps
         poetry_manifest = file_contents.fetch('tool', {}).fetch('poetry', {})
         deps += map_dependencies(poetry_manifest['dependencies'], 'runtime')
+        # Poetry 1.0.0-1.2.0 way of defining dev deps
         deps += map_dependencies(poetry_manifest['dev-dependencies'], 'develop')
+        # Poetry's 1.2.0+ of defining dev deps
+        poetry_manifest
+          .fetch("group", {})
+          .each_pair do |group_name, obj|
+            group_name = "develop" if group_name == "dev"
```

Comment:
I'm familiar with toml and how these are getting parsed. What would the expected group names look like. From the Poetry example what would the `group_name` for `[tool.poetry.group.test.dependencies]` resolve to here?

Reply:
AFAICT it'll use the name that was parsed, so `test` would be the group there. The docs say:

> use a tool.poetry.group.<group> section where <group> is the name of your dependency group (for instance, test):

There's an implicit `main` group that is the default, but we've apparently mapped that to `runtime` until now so I'm setting that as the default in the change below.

Also noteworthy is that anything that's not `main` is considered development semantically, so there's no real distinction between `dev` and `test` to Poetry other than the names:

>  Dependency groups, other than the implicit main group, must only contain dependencies you need in your development process. Installing them is only possible by using Poetry.

And FWIW, I'd prefer to use "development" here to be consistent with the other parsers, but there are 3 uses of "develop" in this file so I'm hesitant to break anything for consumers of the gem : \